test/e2e/nsx_security_policy_test.go

```diff
@@ -186,36 +180,33 @@ func testSecurityPolicyMatchExpression(t *testing.T) {
 	ruleName := "expression-policy-1-rule"
 
 	var err error
-	setupTest(t, ns)
-	defer teardownTest(t, ns, defaultTimeout)
+	testData.createVCNamespace(ns)
+	defer testData.deleteVCNamespace(ns)
 
 	// Create pods
 	podPath, _ := filepath.Abs("./manifest/testSecurityPolicy/allow-client-a-via-pod-selector-with-match-expressions.yaml")
 	require.NoError(t, applyYAML(podPath, ns))
 	defer deleteYAML(podPath, "")
 
-	// Temporarily disable traffic check
-	// clientA := "client-a"
-	// clientB := "client-b"
-	// podA := "pod-a"
-	/*
-		// Wait for pods
-		ps, err := testData.podWaitForIPs(defaultTimeout, clientA, ns)
-		t.Logf("Pods are %v", ps)
-		assert.NoError(t, err, "Error when waiting for IP for Pod %s", clientA)
-		psb, err := testData.podWaitForIPs(defaultTimeout, clientB, ns)
-		t.Logf("Pods are %v", psb)
-		assert.NoError(t, err, "Error when waiting for IP for Pod %s", clientB)
-		iPs, err := testData.podWaitForIPs(defaultTimeout, podA, ns)
-		t.Logf("Pods are %v", iPs)
-		assert.NoError(t, err, "Error when waiting for IP for Pod %s", podA)
-
-		// Ping from pod
-		err = testData.runPingCommandFromPod(ns, clientA, iPs, 4)
-		assert.NoError(t, err, "Error when running ping command from Pod %s", clientA)
-		err = testData.runPingCommandFromPod(ns, clientB, iPs, 4)
-		assert.NoError(t, err, "Error when running ping command from Pod %s", clientB)
-	*/
+	clientA := "client-a"
```

Comment:
Personally I don't prefer to mixed use manifests and kube-client in one case, which is not friendly for both test design and issue troubleshooting. For example, this style requires that the Pod name we used in the test.go must be consistent with manifests, if the dev doesn't know this precondition, a change on the Pod name in test.go may block the cases, and it is not easier to trouble shoot.

Reply:
ok, we will have a discussion before long.